config/src/main/java/org/springframework/security/config/web/util/matcher/RequestMatcherBuilders.java

```diff
@@ -0,0 +1,116 @@
+
+	private static final boolean mvcPresent;
+
+	static {
+		mvcPresent = ClassUtils.isPresent(HANDLER_MAPPING_INTROSPECTOR,
+				AbstractRequestMatcherRegistry.class.getClassLoader());
+	}
+
+	private static final Log logger = LogFactory.getLog(ServletRequestMatcherBuilderFactory.class);
+
+	private RequestMatcherBuilders() {
+
+	}
+
+	/**
+	 * Create the default {@link RequestMatcherBuilder} for use by Spring Security DSLs.
+	 *
+	 * <p>
+	 * If Spring MVC is not present on the classpath or if there is no
+	 * {@link org.springframework.web.servlet.DispatcherServlet}, this method will return
+	 * an Ant-based builder.
+	 *
+	 * <p>
+	 * If the servlet configuration has only
+	 * {@link org.springframework.web.servlet.DispatcherServlet} with a single mapping
+	 * (for example `/` or `/path/*`), then this method will return an MVC-based builder.
+	 *
+	 * <p>
+	 * If the servlet configuration has multiple servlets, but still only one
+	 * {@link org.springframework.web.servlet.DispatcherServlet} with a single mapping
+	 * (for example `/` or `/path/*`), then this method will return a builder that
+	 * determines the request matcher at request time.
+	 *
+	 * <p>
+	 * Otherwise (namely if there are multiple servlet mappings for
+	 * {@link org.springframework.web.servlet.DispatcherServlet}), this with throw an
+	 * exception. In that case, an application should instead use
+	 * {@link ServletRequestMatcherBuilderFactory} to create builders by servlet path.
+	 * @param context the application context
+	 * @return the appropriate {@link RequestMatcherBuilder} based on application
+	 * configuration
+	 */
+	public static RequestMatcherBuilder createDefault(ApplicationContext context) {
+		if (!mvcPresent) {
+			logger.trace("Defaulting to Ant matching since Spring MVC is not on the classpath");
+			return AntPathRequestMatcherBuilder.absolute();
+		}
+		ServletRegistrationCollection registrations = ServletRegistrationCollection.registrations(context);
+		if (registrations.isEmpty()) {
+			logger.trace("Defaulting to MVC matching since Spring MVC is on the class path and no servlet "
+					+ "information is available");
+			return MvcRequestMatcherBuilder.absolute(context);
+		}
+		ServletRegistrationCollection dispatcherServlets = registrations.dispatcherServlets();
+		if (dispatcherServlets.isEmpty()) {
+			logger.trace("Defaulting to MVC matching since there is no DispatcherServlet configured");
+			return AntPathRequestMatcherBuilder.absolute();
+		}
+		ServletPath servletPath = registrations.deduceOneServletPath();
+		if (servletPath != null) {
+			String message = "Defaulting to MVC matching since DispatcherServlet [%s] is the only servlet mapping";
+			logger.trace(String.format(message, servletPath.path()));
+			return MvcRequestMatcherBuilder.relativeTo(context, servletPath.path());
+		}
+		Map<ServletPath, Registration> servletPaths = dispatcherServlets.deduceOnlyServletPaths();
```

Comment:
Shouldn't it be like the following? It seems that every servlet should be mapped, not only `DispatcherServlet`s

```suggestion
		Map<ServletPath, Registration> servletPaths = registrations.deduceOnlyServletPaths();
```